### DISCUSSION TITLE
UI element decoration

### DISCUSSION THREAD

**Discussion by Iaphetes** [bevyengine/bevy#8293] [Q&A]
Hi,

i'm trying to get an ingame UI working with some styling. Like some image in the background of a UI element.
Images added to a NodeBundle are arranged side by side. It would be nice to have some sort of 'background image' possibility.
Additionally it would be nice to have the ability to have different layers to a NodeBundle.

I'm just starting to work with the bevy UI system.
Maybe I'm wrong but it does not seem easily possible at all (I could recreate the entire node structure for the decoration).

Another question:
Is it possible to have elements with relative sizes which have a fixed ratio (e.g. a square button which fills the entire height of the parent)?

Thank you very much in advance.

**Comment by nicopap** [comment]
I don't like at all the bevy layouting system, but it's based on flexbox, so you can use https://the-echoplex.net/flexyboxes/ to figure out the layouts you want.

With regard to background image, I actually don't know, @ickshonpe would know though.

**Reply by Iaphetes** [reply]
Thanks :)
The website looks neat.

I solved/hacked the background image:
Atm  I spawn a parent node with the desired parameters, then two child nodes which fill the entire box. They have to be positioned absolutely to overlap.
I'm not entirely happy with this, but I can do it consisely as a function call, so it doesn't look too bloated
